### ISSUE TITLE
FTL offline after new install on Ubuntu 18.04.2 LTS server (2)

### ISSUE THREAD

**Issue by NLLAPPS** [pi-hole/FTL#609]
Exactly as https://github.com/pi-hole/FTL/issues/562
I have opened a new issue as it was closed

I am using unbound and DNS service does not restart after reboot unless manually started with `sudo service pihole-FTL restart `. I am not sure if it is relevant but I am using Ubuntu default netplan for network config

sudo service pihole-FTL status      
--------------------------------------
Jul 05 20:56:26 fitlet2 systemd[1]: Starting LSB: pihole-FTL daemon...
Jul 05 20:56:26 fitlet2 pihole-FTL[662]: Not running
Jul 05 20:56:26 fitlet2 su[919]: Successful su for pihole by root
Jul 05 20:56:26 fitlet2 su[919]: + ??? root:pihole
Jul 05 20:56:26 fitlet2 su[919]: pam_unix(su:session): session opened for user pihole by (uid=0)
Jul 05 20:56:26 fitlet2 pihole-FTL[662]: dnsmasq: unknown interface eno1
Jul 05 20:56:26 fitlet2 su[919]: pam_unix(su:session): session closed for user pihole
Jul 05 20:56:26 fitlet2 systemd[1]: Started LSB: pihole-FTL daemon.

sudo lsof -i :53
--------------------------------------
Returns nothing

pihole restartdns
--------------------------------------
Does not work

pihole -t
--------------------------------------
Jul  5 20:55:08 dnsmasq[1540]: cached checkipv6.digitalocean.synology.com is 2604:a880:2:d0::21a0:3001
Jul  5 20:55:08 dnsmasq[1540]: cached checkipv6.digitalocean.synology.com is 2604:a880:2:d0::222c:e001
Jul  5 20:55:08 dnsmasq[1540]: cached checkipv6.digitalocean.synology.com is 2604:a880:2:d0::2212:1
Jul  5 20:55:14 dnsmasq[1540]: query[A] ssl.google-analytics.com from 192.168.0.189
Jul  5 20:55:14 dnsmasq[1540]: /etc/pihole/gravity.list ssl.google-analytics.com is 0.0.0.0
Jul  5 20:55:26 dnsmasq[1540]: query[A] fitlet2 from 127.0.0.1
Jul  5 20:55:26 dnsmasq[1540]: /etc/pihole/local.list fitlet2 is 192.168.0.2
Jul  5 20:55:26 dnsmasq[1540]: query[AAAA] fitlet2 from 127.0.0.1
Jul  5 20:55:26 dnsmasq[1540]: config fitlet2 is NODATA-IPv6
**_Notice the SIGTERM here_**
**Jul  5 20:55:26 dnsmasq[1540]: exiting on receipt of SIGTERM**

sudo service pihole-FTL restart 
--------------------------------------
Restarts the service and everything works fine

**Comment by NLLAPPS** [comment]
Can it be related to 
https://bugs.launchpad.net/ubuntu/+source/dnsmasq/+bug/1531184
?

**Comment by DL6ER** [comment]
@NLLAPPS Yes, this is likely. However, the bug you are refering to is about a missing setting in a `systemd` file. I proposed using a `systemd` unit instead of / in addition to the current `init.d` script, however, we were not able to reach consensus and, hence, haven't implemented native `systemd` support so far. Thus, the issue is unrelated altough it is very likely that they are of the same consequence.

> ```
> Jul 05 20:56:26 fitlet2 pihole-FTL[662]: dnsmasq: unknown interface eno1
> Jul 05 20:56:26 fitlet2 su[919]: pam_unix(su:session): session closed for user pihole
> ```

This is the problem - a startup when the network interfaces aren't ready. We can consider adding the ability to FTL to pause some time before starting `dnsmasq`.

> Notice the SIGTERM here
> ```
> Jul 5 20:55:26 dnsmasq[1540]: exiting on receipt of SIGTERM
> ```

This is unrelated. The `SIGTERM` is sent by something/somebody to `pihole-FTL` and it is exiting accordingly. You'd have to find out which process did this (you have likely caused this yourself with `sudo service pihole-FTL stop/restart` or similar).

**Comment by DL6ER** [comment]
@NLLAPPS Did you manually change your configuration? Please generate a debug log and upload it to get a token for us.

---

I tried to replicate your issue with the current Pi-hole by setting
```
interface=ens0
```
in my `/etc/dnsmasq.conf` (this interface does not exist in my device!) but, instead of a fatal failure of `dnsmasq`, I only see
```
Jul  6 16:34:33 dnsmasq[31380]: warning: interface ens0 does not currently exist
```
in the logs and `pihole-FTL` stays up running and accepting queries. It binds to interfaces the wildcard address and discards requests it is configured to not respond to. This has the great advantage of working even when interfaces come and go and change address and is the standard for Pi-hole.

---

The only way I can achieve what you see is when I set
```
interface=ens0
bind-interfaces
```
which should really only ever be needed when running more than one `pihole-FTL`/`dnsmasq` instance on the same machine. Note that, however, this is a configuration we do not fully support as it would need a lot of extra handing whilst being outside of the scope of Pi-hole.

**Comment by NLLAPPS** [comment]
I ma not sure which config you are asking about.

I have nothing but the following line in my /etc/dnsmasq.conf

`conf-dir=/etc/dnsmasq.d`

Contents of /etc/dnsmasq.d folder are:  `01-pihole.conf` and `lxd`

01-pihole.conf has 'interface=eno1' at the bottom. Device has dual LAN and eno1 is not connected, main interface. 'ens0' is. Could that be the issue?

Contents of 01-pihole.conf:

`addn-hosts=/etc/pihole/gravity.list
addn-hosts=/etc/pihole/black.list
addn-hosts=/etc/pihole/local.list

localise-queries

no-resolv

cache-size=10000

log-queries
log-facility=/var/log/pihole.log

local-ttl=2

log-async


dhcp-name-match=set:wpad-ignore,wpad
dhcp-ignore-names=tag:wpad-ignore

localise-queries

no-resolv



cache-size=10000

log-queries
log-facility=/var/log/pihole.log

local-ttl=2

log-async

dhcp-name-match=set:wpad-ignore,wpad
dhcp-ignore-names=tag:wpad-ignore
server=127.0.0.1#5353
server=2606:4700:4700::1111#53
domain-needed
bogus-priv
interface=eno1`

**Comment by DL6ER** [comment]
> Contents of `/etc/dnsmasq.d` folder are: `01-pihole.conf` and `lxd`

There we go: `lxd` should not be there. What is its content?


P.S.: I put your quoted configuration into a code block to make it more readable

**Comment by NLLAPPS** [comment]
# Tell any system-wide dnsmasq instance to make sure to bind to interfaces
# instead of listening on 0.0.0.0
# WARNING: changes to this file will get lost if lxd is removed.
bind-interfaces
except-interface=lxdbr0

**Comment by NLLAPPS** [comment]
ifconfig does not list any interface called `lxdbr0`

**Comment by DL6ER** [comment]
There we have the problem, `bind-interfaces`, just as I mentioned above. Please delete this file (or move it somewhere else outside of this directory) and restart Pi-hole.

Everything should now be in order.

**Comment by NLLAPPS** [comment]
That fixed it. Thank you.
However, I have not created lxd config and this is a fresh installation. What might have created it?

**Comment by DL6ER** [comment]
See https://help.ubuntu.com/lts/serverguide/lxd.html

I haven't seen before that this file gets automatically installed, though.

I will close this issue ticket as we managed to get everything working for you. If you encounter further errors, feel free to either re-open this or create a new ticket, whatever is more appropriate.
